spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
We don't know if it's going to fail or not. I might have that custom table in my database init script. There is no way to know. 

What @vpavic suggested initially is that a user would have to explicitly disable the initializer flag to prevent the exception. Doesn't feel right to me.

It will fail down the road if you configure a custom table name and you have no script to create said custom table name.